lib/restbase.js

```diff
@@ -143,6 +143,9 @@ RESTBase.prototype.request = function (req) {
 RESTBase.prototype._request = function (req) {
     var self = this;
 
+    req.headers = req.headers || {};
+    req.headers['x-request-id'] = req.headers['x-request-id'] || rbUtil.generateRequestId();
```

Comment:
Yeah, but we perform all kinds of internal requests, which aren't always forwarding the headers. That means that different parts of the internal request processing would get their own request ids.